client/src/components/PostEntry.js

```diff
@@ -0,0 +1,113 @@
+import React, {useState} from 'react';
+
+function PostEntry(props) {
+  const [title, setTitle] = useState(props.title || '');
+  const [image, setImage] = useState(props.image || undefined);
+  const [imageSrc, setImageSrc] = useState(props.imageLink || '');
+  const [imageTitle, setImageTitle] = useState(props.imageLink || '');
+  const [link, setLink] = useState(props.link || '');
+  const [subject, setSubject] = useState(props.subject || '');
```

Comment:
I would completely refactor this setup. Instead, pass the `postData` as an optional prop with default values, so that the state can be completely managed by the parent component if this is to be treated as a controlled component.

Although, to be honest, I don't really see a reason for the parent to know about the post state at all. If there is no such reason, just move all of the logic from your wrapper classes into here, and render this component completely conditionally on the `edit` prop. At that point you can just completely get rid of the wrapper classes and encapsulate all of the desired behaviour into this one component.